src/asynk/queries/fetch_task_type.sql

```diff
@@ -0,0 +1 @@
+SELECT * FROM (SELECT * FROM fang_tasks  ORDER BY created_at ASC ) ft WHERE ft.state = 'new' AND ft.task_type = $1 LIMIT 1 FOR UPDATE SKIP LOCKED
```

Comment:
the query seems too complicated. why are you using nested queries?

it can be simplified as:
```sql
SELECT * FROM fang_tasks ORDER BY created_at ASC WHERE state = 'new' AND task_type = $1 LIMIT 1 FOR UPDATE SKIP LOCKED
```
```

Reply:
what about

```rust
SELECT * FROM fang_tasks  WHERE state = 'new' AND task_type = $1 ORDER BY created_at ASC  LIMIT 1 FOR UPDATE SKIP LOCKED

```

Reply:
That works :D changing.